obspy/io/y/core.py

```diff
@@ -256,10 +307,11 @@ def _read_y(filename, headonly=False, **kwargs):  # @UnusedVariable
                 # UCHAR Key[16]
                 #   Unique key that identifies this record in the database.
                 data = fh.read(next_tag)
-                parts = unpack(endian + b'd16s', data[8:])
+                parts = _unpack_with_asciiz_and_decode(
+                    endian + b'd16s', data[8:])
                 params = AttribDict()
                 params.load_date = parts[0]
-                params.key = parts[1].rstrip(b'\x00')
```

Comment:
Yeah. But again, everything that goes into `trace.stats.y` extra headers is untested.